src/Maps/MapWriter.cpp

```diff
@@ -27,7 +27,9 @@ namespace MapWriter {
 
 	void Write(Stream::Writer& streamWriter, const MapData& mapData)
 	{
-		WriteHeader(streamWriter, mapData.header);
+		ValidateMap(mapData);
+
+		streamWriter.Write(mapData.header);
```

Comment:
I like this change. It makes sense to separate the validation of `mapData` from writing of `mapData.header`.